src/apps/statements/tests/getFullActivity/getFullActivity.test.ts

```diff
@@ -91,4 +95,119 @@ describe('getFullActivity', () => {
+    const statement = createStatement({
+      object: TEST_ACTIVITY,
+      ...TEST_CONTEXT_ACTIVITIES,
+    });
+    await service.storeStatements({
+      models: [statement],
+      attachments: [],
+      client: TEST_CLIENT,
+    });
+    const fullActivity = await service.getFullActivity({
+      activityId: TEST_ACTIVITY_ID,
+      client: TEST_CLIENT,
+    });
+    assert.deepEqual(fullActivity, TEST_ACTIVITY_WITH_CONTEXT_ACTIVITIES);
+  });
+
+  it(
+    'should merge the contextActivities when storing two contextActivities in one batch',
+    async () => {
+      const initialStatement = createStatement({
+        object: TEST_ACTIVITY,
+        ...TEST_CONTEXT_ACTIVITIES,
+      });
+      const mergeStatement = createStatement({
+        object: TEST_ACTIVITY,
+        ...TEST_CONTEXT_ACTIVITIES,
+      });
+      await service.storeStatements({
+        models: [initialStatement, mergeStatement],
+        attachments: [],
+        client: TEST_CLIENT,
+      });
+      const fullActivity = await service.getFullActivity({
+        activityId: TEST_ACTIVITY_ID,
+        client: TEST_CLIENT,
+      });
+      assert.deepEqual(fullActivity, TEST_ACTIVITY_WITH_CONTEXT_ACTIVITIES);
+    },
+  );
+
+  it(
+    'should merge two different contextActivities when storing contextActivities in one batch',
+    async () => {
+      const initialStatement = createStatement({
+        object: TEST_ACTIVITY,
+        ...TEST_CONTEXT_ACTIVITIES,
+      });
+      const mergeStatement = createStatement({
+        object: TEST_ACTIVITY,
+        ...ANOTHER_TEST_CONTEXT_ACTIVITIES,
+      });
+      await service.storeStatements({
+        models: [initialStatement, mergeStatement],
+        attachments: [],
+        client: TEST_CLIENT,
+      });
+      const fullActivity = await service.getFullActivity({
+        activityId: TEST_ACTIVITY_ID,
+        client: TEST_CLIENT,
+      });
+      assert.deepEqual(fullActivity, TEST_ACTIVITY_WITH_MERGED_CONTEXT_ACTIVITIES);
+    },
+  );
+
+  it(
+    'should return last contextActivities when storing contextActivities in two batches',
+    async () => {
+      const initialStatement = createStatement({
+        object: TEST_ACTIVITY,
+        ...TEST_CONTEXT_ACTIVITIES,
+      });
+      const mergeStatement = createStatement({
+        object: TEST_ACTIVITY,
+        ...ANOTHER_TEST_CONTEXT_ACTIVITIES,
+      });
+      await service.storeStatements({
+        models: [initialStatement],
+        attachments: [],
+        client: TEST_CLIENT,
+      });
+      await service.storeStatements({
+        models: [mergeStatement],
+        attachments: [],
+        client: TEST_CLIENT,
+      });
+      const fullActivity = await service.getFullActivity({
+        activityId: TEST_ACTIVITY_ID,
+        client: TEST_CLIENT,
+      });
+      assert.deepEqual(fullActivity, TEST_ACTIVITY_WITH_MERGED_CONTEXT_ACTIVITIES);
+    },
+  );
+// tslint:disable-next-line:max-file-line-count
```

Comment:
@PrinceWaune we don't use such ignore lines in other modules within the xapi-service - is this required?